src/user/return-find-user.dto.ts

```diff
@@ -0,0 +1,34 @@
+import { BaseEntity } from "../common";
+
+export const PROFILE_PROVIDER = {
+  GOOGLE: "google",
+  APPLE: "apple",
+  KAKAO: "kakao",
+  NAVER: "naver",
+  ETC: "etc",
+} as const;
+
+export const PROFILE_PROFILE_ARRAY = Object.values(PROFILE_PROVIDER);
+export type ProfileProvider = typeof PROFILE_PROFILE_ARRAY[number];
```

Comment:
```suggestion
type ValueOf<T> = T[keyof T];
export type ProfileProvider = ValueOf<typeof PROFILE_PROVIDER>;

```
Array 값 자체가 필요한게 아니라면 이렇게 작성해도 괜찮을 것 같은데 값 자체가 필요한 걸까요?

Reply:
백 레포의 UserProfile Entity enum 값 으로 지정하려고 만들어 놓은 것 인데
프론트에서도 enum 사용하시지 않을까 해서 넣어 두었습니다.

생각해보니 클라에서 회원가입 할 때 따로 활용 할 것 같지는 않아보이네요.
위에 적어 놓은 것으로 바꿔 놓겠습니다 🌝